dm/syncer/data_validator.go

```diff
@@ -192,11 +194,12 @@ type DataValidator struct {
 	streamerController *binlogstream.StreamerController
 	persistHelper      *validatorPersistHelper
 
-	validateInterval time.Duration
-	checkInterval    time.Duration
-	cutOverLocation  *binlog.Location
-	workers          []*validateWorker
-	workerCnt        int
+	validateInterval      time.Duration
+	checkInterval         time.Duration
+	cutOverLocationAtomic atomic.Pointer[binlog.Location]
```

Comment:
```suggestion
	cutOverLocation atomic.Pointer[binlog.Location]
```

prefer this, `Atomic` is excessive.